llvm/lib/Target/WebAssembly/Disassembler/WebAssemblyDisassembler.cpp

```diff
@@ -289,6 +289,24 @@ MCDisassembler::DecodeStatus WebAssemblyDisassembler::getInstruction(
         return MCDisassembler::Fail;
       break;
     }
+    case WebAssembly::OPERAND_CATCH_LIST: {
+      if (!parseLEBImmediate(MI, Size, Bytes, false))
+        return MCDisassembler::Fail;
+      int64_t NumCatches = MI.getOperand(MI.getNumOperands() - 1).getImm();
```

Comment:
where do these operands actually get added to the MI?

Reply:
In those `parse***` functions. For example:
https://github.com/llvm/llvm-project/blob/6ca5c397a99aca2b9faa9b2d96238c5bda33010e/llvm/lib/Target/WebAssembly/Disassembler/WebAssemblyDisassembler.cpp#L101-L108

Yeah it was a little confusing to me too so I had to look into those functions...

Reply:
Ah, I see, the number of catches is the last operand currently, because we haven't parsed the rest yet.